packages/workspace/src/browser/workspace-delete-handler.ts

```diff
@@ -0,0 +1,131 @@
+/********************************************************************************
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { ConfirmDialog, ApplicationShell, Navigatable, SaveableWidget, Saveable, Widget } from '@theia/core/lib/browser';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import { UriCommandHandler } from '@theia/core/lib/common/uri-command-handler';
+import { WorkspaceService } from './workspace-service';
+
+@injectable()
+export class WorkspaceDeleteHandler implements UriCommandHandler<URI[]> {
+
+    @inject(FileSystem)
+    protected readonly fileSystem: FileSystem;
+
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    @inject(WorkspaceService)
+    protected readonly workspaceService: WorkspaceService;
+
+    isVisible(uris: URI[]): boolean {
+        if (!uris.length) {
+            return false;
+        }
+        const rootUris = this.workspaceService.tryGetRoots().map(root => new URI(root.uri));
+        return !rootUris.some(rootUri => uris.some(uri => uri.isEqualOrParent(rootUri)));
```

Comment:
Looks like the command would be visible if one of the uris has one of the roots as parent. 
Didn't you want to check all, i.e.
```ts
return uris.every(uri => rootUris.some(rootUri => uri.isEqualOrParent(rootUri)));
```

Reply:
Do you have any cases in mind, where a URI would not be a child of a workspace root? And if so why would you want to disable the command?

Reply:
> Looks like the command would be visible if one of the uris has one of the roots as parent.

`uri.isEqualOrParent(rootUri)` - it checks that no root will be deleted, not that a file should belong to a root.

Reply:
Maybe it's worth have a short inline comment about the intend as it wasn't clear (to me at least :)).